api/src/main/java/net/kyori/adventure/translation/TranslationRegistry.java

```diff
@@ -131,31 +140,27 @@ public interface TranslationRegistry extends Translator {
    *
    * @param locale a locale
    * @param formats a map of translation keys to formats
-   * @throws IllegalArgumentException if a translation key is already exists
+   * @throws IllegalArgumentException if a translation key already exists
    * @see #register(String, Locale, MessageFormat)
    * @since 4.0.0
+   * @deprecated For removal since 4.20.0. Use {@link TranslationStore#messageFormat(Key)} instead.
    */
-  default void registerAll(final @NotNull Locale locale, final @NotNull Map<String, MessageFormat> formats) {
-    this.registerAll(locale, formats.keySet(), formats::get);
-  }
+  @Deprecated
+  void registerAll(final @NotNull Locale locale, final @NotNull Map<String, MessageFormat> formats);
 
   /**
    * Registers a resource bundle of translations.
    *
    * @param locale a locale
    * @param path a path to the resource bundle
    * @param escapeSingleQuotes whether to escape single quotes
-   * @throws IllegalArgumentException if a translation key is already exists
+   * @throws IllegalArgumentException if a translation key already exists
    * @see #registerAll(Locale, ResourceBundle, boolean)
    * @since 4.0.0
+   * @deprecated For removal since 4.20.0. Use {@link TranslationStore#messageFormat(Key)} instead.
    */
-  default void registerAll(final @NotNull Locale locale, final @NotNull Path path, final boolean escapeSingleQuotes) {
```

Comment:
I think removing these default implementations is technically a breaking change?

Reply:
Yes you're right, I'll read these